src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/PollLogsStrategy.cs

```diff
@@ -97,18 +115,76 @@ public void Notify(IStorageBlob blobWritten)
                     break;
                 }
 
-                await NotifyRegistrationsAsync(blob, cancellationToken);
+                notifications.Add(NotifyRegistrationsAsync(blob, failedNotifications, cancellationToken));
```

Comment:
What are the bounds on the number of things that can be in the background queue? While we want to parallelize things, we also need to constrain things so that we're not attempting to start thousands of tasks simultaneously.